### PR TITLE
Add coder-on-railway blog post

### PR DESCRIPTION
**Pull request by matifali**
This PR adds a blog post to promote our Railway template for Coder.

https://railway.app/template/cUQ8_P?referralCode=tfH8Uw

[Deploy on Railway](https://railway.app/template/cUQ8_P?referralCode=tfH8Uw)
Merging this will close https://github.com/coder/coder/issues/6861

### CODE REVIEW

**Comment by matifali** [comment]
After you approve, I can publish on CMS too. Your call

File: posts/coder-on-railway/index.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+[![Deploy on Railway](https://railway.app/button.svg)](https://railway.app/template/cUQ8_P?referralCode=tfH8Uw)
+
+### 1. Fill the environment variables
+
+The current version of the template allows you to set an optional environment variable `CODER_WILDCARD_ACCESS_URL` which will allow you to forward ports from your workspace to a custom wildcard subdomain of the URL you provide. This is useful if you want to access a service running in your workspace from a browser. For example, if you set `CODER_WILDCARD_ACCESS_URL` to `*.coder.example.com`, you can access a service running on port 8080 in your workspace by visiting `8080--<username>--<workspace-name>.coder.example.com`.
+
+### 2. Attach a custom domain
+
+Your Coder deployment will always be accessible at `https://<app-name>.up.railway.app`. If you want to use a custom domain, you can do so by going to Dashboard > Coder (Service) > Settings > Domains and adding your domain and optionally a wildcard subdomain if you specified `CODER_WILDCARD_ACCESS_URL` in the previous step.
```
Comment (bpmct) [inline posts/coder-on-railway/index.md:17]: If you think it'd be helpful, you can link to our docs here: https://coder.com/docs/v2/latest/admin/configure#wildcard-access-url

File: posts/coder-on-railway/index.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+### Official
+
+- [**VS Code Extension**](https://marketplace.visualstudio.com/items?itemName=coder.coder-remote): Open any Coder workspace in VS Code with a single click
+- [**JetBrains Gateway Extension**](https://plugins.jetbrains.com/plugin/19620-coder): Open any Coder workspace in JetBrains Gateway with a single click
+
+### Community
+
+- [**Coder GitHub Action**](https://github.com/marketplace/actions/update-coder-template): A GitHub Action that updates Coder templates
+- [**Various Templates**](https://github.com/coder/coder/examples/templates/community-templates.md): Hetzner Cloud, Docker in Docker, and other templates the community has built.
```
Comment (bpmct) [inline posts/coder-on-railway/index.md:61]: Can you change these into bullet points? It feels a bit strange to have these dedicated sections for a blog post

File: posts/coder-on-railway/index.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# Deploy Coder on Railway.app
+
+Coder can be deployed on Railway.app in just a few clicks. This guide will walk you through the process of deploying Coder on Railway.app.
```
[the remark this answers is not in the document]
Comment (matifali) [reply posts/coder-on-railway/index.md:3]: I agree you are right. We should assume that user doesn't know what is Coder.

File: posts/coder-on-railway/index.md (hunk 2 above)
Comment (matifali) [reply posts/coder-on-railway/index.md]: done It was not looking good,

File: posts/coder-on-railway/index.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+## Quickstart
+
+Start your deployment by clicking the button below. This will take you to Railway.app where you can sign up for a free account and deploy Coder.
+
+[![Deploy on Railway](https://railway.app/button.svg)](https://railway.app/template/cUQ8_P?referralCode=tfH8Uw)
+
+### 1. Fill the environment variables
+
+The current version of the template allows you to set an optional environment variable `CODER_WILDCARD_ACCESS_URL` which will allow you to forward ports from your workspace to a custom wildcard subdomain of the URL you provide. This is useful if you want to access a service running in your workspace from a browser. For example, if you set `CODER_WILDCARD_ACCESS_URL` to `*.coder.example.com`, you can access a service running on port 8080 in your workspace by visiting `8080--<username>--<workspace-name>.coder.example.com`.
```
[the remark this answers is not in the document]
Comment (matifali) [reply posts/coder-on-railway/index.md]: done

**Pull request merged**